src/flashbar/index.tsx

```diff
@@ -23,62 +25,142 @@ export default function Flashbar({ items, ...restProps }: FlashbarProps) {
   const [breakpoint, ref] = useContainerBreakpoints(['xs']);
   const isRefresh = useVisualRefresh();
   const baseProps = getBaseProps(restProps);
-
   const mergedRef = useMergeRefs(ref, __internalRootRef);
 
-  // All the flash items should have ids so we can identify which DOM element is being removed from the DOM to animate it.
+  /**
+   * All the flash items should have ids so we can identify which DOM element is being
+   * removed from the DOM to animate it. Motion will be disabled if any of the provided
+   * flash messages does not contain an `id`.
+   */
   const motionDisabled = useReducedMotion(ref as any) || !isRefresh || (items && !items.every(item => 'id' in item));
-  // Motion will be disabled if any of the provided flash messages does not contain an `id`
 
-  const renderFlashItem = (item: FlashbarProps.MessageDefinition, index: number) => (
-    <Flash
-      key={item.id ?? index}
-      // eslint-disable-next-line react/forbid-component-props
-      className={clsx(isRefresh ? styles['flash-refresh'] : '')}
-      {...item}
-    />
-  );
-  const renderFlashItemsWithTransitions = (items: readonly FlashbarProps.MessageDefinition[]) => {
+  /**
+   * The `enableStackingOption` property is a hidden boolean that allows for teams
+   * to beta test the flashbar stacking feature. This property will add an additional
+   * toggle control into the flashbar that will change the state of the `isFlashbarStacked`
+   * boolean value by a given user.
+   */
+  const enableStackingOption = (restProps as any).enableStackingOption;
+  const [isFlashbarStacked, setIsFlashbarStacked] = useState(true);
+
+  /**
+   * If the stacking feature is enabled and there is more than one item in the flashbar then
+   * the toggle will be rendered to change the state from flat to stacked.
+   */
+  function renderStackingOption() {
+    if (!enableStackingOption || !items || items?.length < 2) {
+      return;
+    }
+
     return (
-      <>
-        <TransitionGroup component={null}>
-          {items &&
-            items.map((item, index) => (
-              <Transition
-                transitionChangeDelay={{ entering: TIMEOUT_FOR_ENTERING_ANIMATION }}
+      <div className={clsx(styles['stacking-option'])}>
```

Comment:
I don't see the toggle when opening `http://localhost:8080/#/light/flashbar/stacking/` , how do I activate it?

Reply:
@abdhalees I just re-requested review. There have been significant changes from the first revision. For testing you can use any demo page that has notifications and turn on the secret API by adding `{...{ enableStackingOption: true }}` to the Flashbar component. Some good pages to test with are in the `AppLayout` and `Flashbar` sections.